src/main/java/com/gmail/goosius/siegewar/settings/SiegeWarSettings.java

```diff
@@ -763,19 +766,42 @@ private static void addSpecialEffects(ItemStack artefact, ItemMeta itemMeta, Str
         }
 		//Add special effects
 		for(String[] effectSpec: effectSpecs) {
+		    if (!validateEffect(material, effectSpec[0]))
+		        return false;
 			addSpecialEffect(material, itemMeta, effectSpec);
 		}
 		//Set updated item meta
 		artefact.setItemMeta(itemMeta);
+		return true;
 	}
 
-	private static void addSpecialEffect(Material material, ItemMeta itemMeta, String[] effectSpec) {
-		if(effectSpec[0].equalsIgnoreCase("custom_effect")) {
-			addCustomEffect(itemMeta, effectSpec);
-		} else if(material == Material.POTION
+	// Make sure that the config has been given a valid PotionEffectType or Enchantment name.
+    private static boolean validateEffect(Material material, String name) {
+        return !name.isEmpty()
+                && name.equals("custom_effect") // ignore custom_effect
+                && ((isPotionBased(material) && !validatePotionEffectType(name))
+                    || !isPotionBased(material) && !validateEnchantmentType(name));

```

Comment:
🚧 
- This logic is now problematic in a new way - it returns true only if an effect is both custom AND vanilla. Which will never be true.
- Rather, it should return true if an effect is custom OR vanilla.